pkg/lib/config/oauth.go

```diff
@@ -5,7 +5,11 @@ var _ = Schema.Add("OAuthConfig", `
 	"type": "object",
 	"additionalProperties": false,
 	"properties": {
-		"clients": { "type": "array", "items": { "$ref": "#/$defs/OAuthClientConfig" } }
+		"clients": {
+			"type": "array",
+			"items": { "$ref": "#/$defs/OAuthClientConfig" },
+			"maxItems": 50
+		}
```

Comment:
Changing this may break existing projects. I suggest we apply the limit when authgear.yaml is updated. So it only breaks when someone tries to add more than 50 clients.